fix: auto-delete empty field in Kanban Board

AmanNegi opened the pull request:
<!---
Thank you for submitting a pull request to AppFlowy. The team will dedicate their best efforts to reviewing and approving your pull request. If you have any questions about the project or feedback for us, please join our [Discord](https://discord.gg/wdjWUXXhtw).
-->

<!---
If your pull request adds a new feature, please drag and drop a video into this section to showcase what you've done! If not, you may delete this section.
-->

### Feature Preview
Now the field in Kanban board item is automatically deleted when the user tries to save an empty field.

> Issue: #1898 

https://user-images.githubusercontent.com/37607224/230738649-95b5a008-2a46-4acc-ac1c-769a4fccf662.mp4


#### PR Checklist

- [✅] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [✅] I've listed at least one issue that this PR fixes in the description above.
- [ - ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [✅] All existing tests are passing.

a-wallen commented:
Please don't check off the ` I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.` box if you haven't added any tests :)

a-wallen commented on frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/board_page.dart:
```diff
@@ -282,6 +284,26 @@ class _BoardContentState extends State<BoardContent> {
               .add(BoardEvent.endEditingRow(groupItem.row.id));
+
+          /// Check if the field contains some text
+          var fieldID = fieldController.fieldInfos[0].id;
```
Shouldn't check if the `fieldInfos` is empty before you access an index?

AmanNegi replied:
Sure, I will add an `isEmpty` check as well.

AmanNegi commented:
> Please don't check off the ` I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.` box if you haven't added any tests :)

Sorry for that, I have updated the checklist now.

annieappflowy commented:
@AmanNegi , thanks for contributing. After reviewing this PR, I don't think it's a good improvement from the user's perspective. "Untitled" should be a valid input and not incur data loss, so I will close the PR. Please let me know if you need more clarification.

AmanNegi commented:
Okay if you think it's not appropriate, it's ok. Thanks for the clarification! :smile:
